proof/verifier.go

```diff
@@ -511,7 +527,12 @@ type GroupAnchorVerifier func(gen *asset.Genesis,
 func (p *Proof) Verify(ctx context.Context, prev *AssetSnapshot,
 	headerVerifier HeaderVerifier, merkleVerifier MerkleVerifier,
 	groupVerifier GroupVerifier,
-	chainLookup asset.ChainLookup) (*AssetSnapshot, error) {
+	chainLookup asset.ChainLookup,
+	opts ...ProofVerificationOption) (*AssetSnapshot, error) {
+
+	for _, opt := range opts {
```

Comment:
See comment above. This wouldn't actually modify the `Proof` struct but an instance of `proofVerificationParams` that stores the values. Then we can pass in `params.ChallengeBytes` below to `p.verifyChallengeWitness`.